sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/Metrics.java

```diff
@@ -96,6 +97,21 @@ public static Gauge gauge(Class<?> namespace, String name) {
     return new DelegatingGauge(MetricName.named(namespace, name));
   }
 
+  public static class Internal {
+    public static Counter counter(String namespace, String name, boolean processWideContainer) {
+      return new DelegatingCounter(MetricName.named(namespace, name), processWideContainer);
+    }
+
+    public static Histogram histogram(
+        String namespace,
```

Comment:
namespace and name parameters are specific to user metric (Which is a metric that populates a specific MonitoringInfo URN with a name and namspace parameter).

Have this just take a MetricName instead which allows for non user metrics
I.e. like the ElementCount metric
which is uniquely identifiers by a URN and labels.

https://github.com/apache/beam/pull/7272/files#diff-3834ef03b753005ee8251870a88fe3505732e9a7ad34da600f78e33ab9f37139R45

You can make it under the LabeledMetrics class, which is meant to be an internal implementation for SDK harness implementations. Not part of the public API (Much like the internal.Metrics in python you've added).

Reply:
Done. Moved into `LabeledMetrics`.